crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -583,7 +590,7 @@ impl TypingImporter<'_, '_> {
 
 impl SemanticSyntaxContext for Checker<'_> {
     fn python_version(&self) -> PythonVersion {
-        self.target_version
```

Comment:
It would be great to have a comment here explaining why we use `latest` instead of default. It's very subtle